packages/@aws-cdk/aws-apigatewayv2/lib/common/integration.ts

```diff
@@ -9,4 +9,36 @@ export interface IIntegration extends IResource {
    * @attribute
    */
   readonly integrationId: string;
-}
\ No newline at end of file
+}
+
+/**
+ * AWS service integration sub types
+ * https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-aws-services-reference.html
+ */
+export enum AwsServiceIntegrationSubtype {
+  EVENT_BRIDGE_PUT_EVENTS = 'EventBridge-PutEvents'
+  // AppConfig-GetConfiguration
+  // Kinesis-PutRecord
+  // SQS-DeleteMessage
+  // SQS-PurgeQueue
+  // SQS-ReceiveMessage
+  // SQS-SendMessage
+  // StepFunctions-StartExecution
+  // StepFunctions-StartSyncExecution
+  // StepFunctions-StopExecution
+}
+
+/**
+ * Integration request parameters for EventBridge
+ */
+export interface EventBridgeIntegrationRequestParameters {
+  readonly detail?: string;
+  readonly detailType?: string;
+  readonly eventBusName?: string;
+  readonly region?: string;
+  readonly resources?: string[];
+  readonly source?: string;
+  readonly time?: string;
+}
```

Comment:
These look specific to event bridge. Could we add these directly into `EventBridgeIntegrationProps`?

Also, what would be the defaults for these optional properties?

Reply:
This might be problematic as HttpRouteIntegrationConfig would need to incorporate a flat list of every available AWS service API request parameter for each service that is implemented. This is why by the time CfnIntegration is called, which expects requestParameters in CFN format, there is a private renderRequestParameters method to do just that based on the AWS service integration type, which is in turn why EventBridgeIntegrationRequestParameters exists.

So if we implement StepFunctions integrations next, the thought is requestParameters at the L2 level is of type:

EventBridgeIntegrationRequestParameters | 
  StepFunctionsStartExecutionRequestParameters |
  StepFunctionsStartSyncExecutionRequestParameters |
  StepFunctionsStopExecutionRequestParameters

So we can keep the L1 implementation the same for any future AWS service integration and its corresponding request parameters.

Maybe I'm misunderstanding your ask. Can you clarify?

Reply:
(Originally this is why the type alias AwsServiceIntegrationRequestParameters was used... to avoid the exploding union type)

Reply:
Unfortunately, we don't support union types in jsii.

How about we just model `requestParameters` either as type `any` or a `{ [key: string]: any }` in `HttpRouteIntegrationConfig` and `HttpIntegrationProps`?
Explicit modeling will be applied to the specific subtypes, viz. `EventBridgeIntegration`.

Reply:
Foiled again. OK sounds good.

Reply:
😊

In case you're interested - https://github.com/aws/aws-cdk-rfcs/pull/194

Reply:
Interesting OK. I'll review further and at least understand the intent to maximize consistency across all of the jsii target languages.

Does this mean we should avoid [TypeScript utility types](https://www.typescriptlang.org/docs/handbook/utility-types.html) as well?

Reply:
In thinking of cases where:

```
type Record<K extends string, T> = {
  [P in K]: T
}
```

might be used for

```
requestParameters: Record<string, any> // in context of aws-apigatewayv2::HttpIntegration
// becoming
requestParameters: Record<'detail' | 'detailType' | ... | 'time', any> // shape of EventBridgeIntegrationRequestParameters
```

Reply:
> Does this mean we should avoid TypeScript utility types as well?

Yes. I would recommend sticking to standard types.

Reply:
It's really hard to replicate these fancy type mappings in other languages, and future languages we wish to expand into.

Reply:
Understood. I do notice various sections of existing CDK code making use of Record<string, T>, though. Updated to use a primitive object definition with [key: string] per your suggestion.